### PR TITLE
NEXUS-8990 - Publishing to a hosted nuget repo affects the count of components in a proxy repo

### PR DESCRIPTION
**Pull request by kellyrob99**
- only execute count query if hosted repos are in context

https://issues.sonatype.org/browse/NEXUS-8990
http://bamboo.s/browse/NX3-OSSF595

### CODE REVIEW

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java [hunk 1]
```diff
@@ -128,12 +128,17 @@ protected void doDestroy() throws Exception {
 
   @Override
   public int count(final String operation, final Map<String, String> parameters) {
+    Iterable<Repository> proxyRepositories = getProxyRepositories();
     final List<Integer> remoteCounts = passQueryToRemoteRepos(nugetQuery(operation, parameters),
-        getProxyRepositories(),
+        proxyRepositories,
         new CountFetcher(fetcher));
-
-    final int hostedCount = count(operation, parameters, getHostedRepositories());
```
Comment (kellyrob99) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java]: I'm sure this was just unexpected, but passing in no repositories here resulted in a blanket query for all components

Comment (mcculls) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java]: I wonder if we should change the queries in `StorageTxImpl` to return no results (or count 0) when no repositories are given? Is there ever a situation where passing no repositories into `countAssets`, etc. should query over everything?

Comment (mcculls) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java]: Actually to be fair, the `StorageTx` javadoc does mention this for the `findAssets` case:

https://github.com/sonatype/nexus-oss/blob/master/components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTx.java#L116

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java [hunk 2]
```diff
@@ -128,12 +128,17 @@ protected void doDestroy() throws Exception {
...
-
+    log.debug("Counts from proxyRepositories: {} are: {}", proxyRepositories, remoteCounts);
+    
+    Iterable<Repository> hostedRepositories = getHostedRepositories();
+    final int hostedCount = hostedRepositories.iterator().hasNext()
```
Comment (mcculls) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java]: you could also use `!Iterables.isEmpty(...)` which might be slightly more readable

**Comment by mcculls** [comment]
+1 good catch

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
+1

**Pull request closed without merging**